### ISSUE TITLE
Error when running yarn ios using starters

### ISSUE THREAD

**Issue by tkhg** [tamagui/tamagui#2279]
### Current Behavior

using `npm create tamagui@latest --template expo-router` and follow the steps to create a project, then run `yarn ios` to start development via  expo go or expo-dev-client, the same error is encountered in both ways
```
 ERROR  ReferenceError: Property 'document' doesn't exist, js engine: hermes
    at ContextNavigator (http://localhost:8081/node_modules/expo-router/entry.bundle//&platform=ios&dev=true&lazy=true&minify=false&inlineSourceMap=false&modulesOnly=false&runModule=true&app=com.anonymous.expo-router-example&transform.routerRoot=app&transform.engine=hermes:148411:24)
    at ExpoRoot (http://localhost:8081/node_modules/expo-router/entry.bundle//&platform=ios&dev=true&lazy=true&minify=false&inlineSourceMap=false&modulesOnly=false&runModule=true&app=com.anonymous.expo-router-example&transform.routerRoot=app&transform.engine=hermes:148367:28)
    at App
    at ErrorToastContainer (http://localhost:8081/node_modules/expo-router/entry.bundle//&platform=ios&dev=true&lazy=true&minify=false&inlineSourceMap=false&modulesOnly=false&runModule=true&app=com.anonymous.expo-router-example&transform.routerRoot=app&transform.engine=hermes:341210:24)
    at ErrorOverlay
    at withDevTools(ErrorOverlay) (http://localhost:8081/node_modules/expo-router/entry.bundle//&platform=ios&dev=true&lazy=true&minify=false&inlineSourceMap=false&modulesOnly=false&runModule=true&app=com.anonymous.expo-router-example&transform.routerRoot=app&transform.engine=hermes:340744:27)
    at RCTView
    at View (http://localhost:8081/node_modules/expo-router/entry.bundle//&platform=ios&dev=true&lazy=true&minify=false&inlineSourceMap=false&modulesOnly=false&runModule=true&app=com.anonymous.expo-router-example&transform.routerRoot=app&transform.engine=hermes:40740:43)
    at RCTView
    at View (http://localhost:8081/node_modules/expo-router/entry.bundle//&platform=ios&dev=true&lazy=true&minify=false&inlineSourceMap=false&modulesOnly=false&runModule=true&app=com.anonymous.expo-router-example&transform.routerRoot=app&transform.engine=hermes:40740:43)
    at AppContainer (http://localhost:8081/node_modules/expo-router/entry.bundle//&platform=ios&dev=true&lazy=true&minify=false&inlineSourceMap=false&modulesOnly=false&runModule=true&app=com.anonymous.expo-router-example&transform.routerRoot=app&transform.engine=hermes:40551:36)
    at main(RootComponent) (http://localhost:8081/node_modules/expo-router/entry.bundle//&platform=ios&dev=true&lazy=true&minify=false&inlineSourceMap=false&modulesOnly=false&runModule=true&app=com.anonymous.expo-router-example&transform.routerRoot=app&transform.engine=hermes:124413:28)
 ```
 and then a same error is repeated continuously
 ```
λ Bundled 6992ms (node_modules/expo-router/node/render.js)
Web Bundled 7092ms (node_modules/expo-router/entry.js)
λ  WARN  props.pointerEvents is deprecated. Use style.pointerEvents
λ  WARN  "shadow*" style props are deprecated. Use "boxShadow".
  styleq (node_modules/styleq/dist/styleq.js:62:39)
λ  WARN  accessibilityRole is deprecated. Use role.
λ  WARN  accessibilityLabel is deprecated. Use aria-label.
λ  WARN  Image: style.resizeMode is deprecated. Please use props.resizeMode.
λ  WARN  Image: style.tintColor is deprecated. Please use props.tintColor.
 ERROR  There was a problem connecting to the react-native-css-interop Metro server
```
<img width="514" alt="image" src="https://github.com/tamagui/tamagui/assets/28704185/cbfae5bd-40e3-464b-9df2-5ca087eb6048">

### Expected Behavior

start the project on expo go or expo-dev-client without error

### Tamagui Version

```markdown
1.89.26
```

### Platform (Web, iOS, Android)

```markdown
iOS
```

### Reproduction

```markdown
Expo go steps:
1. run `npm create tamagui@latest --template expo-router`
2. select `Expo Router (beta) - Expo Router starter with Tamagui set up`
3. `cd` into the project and run `yarn ios`
4. press `i` to expo-go app in iOS simulator

Dev build step
1. run `npm create tamagui@latest --template expo-router`
2. select `Expo Router (beta) - Expo Router starter with Tamagui set up`
3. run `npx expo run:ios` to create a dev build
4. upon build succeeded, press `i` to launch dev build in iOS simulator
```

### System Info

```markdown
System:
    OS: macOS 14.2.1
    CPU: (10) arm64 Apple M1 Pro
    Memory: 4.32 GB / 32.00 GB
    Shell: 5.9 - /bin/zsh
  Binaries:
    Node: 21.1.0 - ~/.nvm/versions/node/v21.1.0/bin/node
    Yarn: 4.0.2 - ~/.nvm/versions/node/v21.1.0/bin/yarn
    npm: 10.2.0 - ~/.nvm/versions/node/v21.1.0/bin/npm
    pnpm: 8.14.1 - /opt/homebrew/bin/pnpm
    bun: 1.0.26 - ~/.bun/bin/bun
    Watchman: 2023.12.04.00 - /opt/homebrew/bin/watchman
  Browsers:
    Brave Browser: 119.1.60.118
    Chrome: 121.0.6167.184
    Safari: 17.2.1
  npmPackages:
    @babel/core: ^7.23.3 => 7.23.9
    @expo/metro-runtime: ~3.1.3 => 3.1.3
    @react-navigation/native: ^6.0.2 => 6.1.10
    @tamagui/config: ^1.89.26 => 1.89.26
    @tamagui/metro-plugin: ^1.89.26 => 1.89.26
    @types/react: ~18.2.14 => 18.2.55
    babel-preset-expo: ^10.0.1 => 10.0.1
    expo: ~50.0.6 => 50.0.6
    expo-font: ~11.10.2 => 11.10.2
    expo-linking: ~6.2.1 => 6.2.2
    expo-router: ~3.4.7 => 3.4.7
    expo-splash-screen: ~0.26.4 => 0.26.4
    expo-status-bar: ~1.11.1 => 1.11.1
    expo-system-ui: ~2.9.3 => 2.9.3
    expo-web-browser: ~12.8.2 => 12.8.2
    react: ^18.2.0 => 18.2.0
    react-dom: ^18.2.0 => 18.2.0
    react-helmet-async: ~2.0.4 => 2.0.4
    react-native: 0.73.4 => 0.73.4
    react-native-reanimated: ~3.6.2 => 3.6.2
    react-native-safe-area-context: 4.8.2 => 4.8.2
    react-native-screens: ~3.29.0 => 3.29.0
    react-native-web: ^0.19.9 => 0.19.10
    tamagui: ^1.89.26 => 1.89.26
    typescript: ^5.3.3 => 5.3.3
```

**Comment by ifiokjr** [comment]
I'm having this same issue. Are there currently any workarounds?

**Comment by youssdevx** [comment]
Same here
<img width="598" alt="image" src="https://github.com/tamagui/tamagui/assets/21293851/e1ebaf82-2a56-4bbf-b536-5c45bf15da13">

**Comment by joaolfern** [comment]
>  ERROR  ReferenceError: Property 'document' doesn't exist, js engine: hermes

Running into this on `yarn android` too

**Comment by TheEhsanSarshar** [comment]
I'll check that over weekend

**Comment by youssdevx** [comment]
Upgrading to 1.90.6 resolved my issue

**Comment by TheEhsanSarshar** [comment]
I'm not able to reproduce it, can you try the latest version and let me know if the issue persists

**Comment by kasparas-ban** [comment]
@TheEhsanSarshar I created a new Expo project with `npx create-expo-app --template` (File based router), then followed the instructions on https://tamagui.dev/docs/guides/expo to add Tamagui to the project, and I'm getting the same endless stream of `There was a problem connecting to the react-native-css-interop Metro server` in the console.

Here's my example project: https://github.com/kasparas-ban/tamagui-test

**Comment by TheEhsanSarshar** [comment]
@kasparas-ban on native?

**Comment by kasparas-ban** [comment]
@TheEhsanSarshar Yes. I've tested this in every which way I could think of: on native, on web, on iOS and Android, using yarn and npm. Still the same error. I'm a bit surprised that not more people are getting this. I've used nothing but the official template and the exact instructions from the docs. Are you able to reproduce it?

**Comment by giornn0** [comment]
> @TheEhsanSarshar Yes. I've tested this in every which way I could think of: on native, on web, on iOS and Android, using yarn and npm. Still the same error. I'm a bit surprised that not more people are getting this. I've used nothing but the official template and the exact instructions from the docs. Are you able to reproduce it?
I got the same problem, with template created 2 days ago.

My workaround was to dynamically import the css in app/_layout.tsx:
`
import { Platform } from "react-native";

if (Platform.OS === "web") {
  import("../tamagui-web.css");
}
`
And the application seems to work fine in android. At least the error doesn't show.
But I don't know how this impact in tamagui correct functionality.

**Comment by TheEhsanSarshar** [comment]
@kasparas-ban I was able to reproduce with your repo `https://github.com/kasparas-ban/tamagui-test` 
but when I create a new repo using `npm create tamagui@latest` things are working fine.
your repository seems to have some missing dependencies like `reanimated`

**Comment by kasparas-ban** [comment]
@TheEhsanSarshar  I'm guessing you were testing using yarn. I did notice a weird quirk that yarn misses reanimated although npm works fine. Regardless, if I install reanimated manually I get the same error. Also, I can verify that new projects created via `npm create tamagui@latest` work fine for me as well. It's when I try to add Tamagui to an existing Expo project that's failing.

I can confirm that @giornn0 solution does work.

**Comment by TheEhsanSarshar** [comment]
if you want to add Tamagui to an existing project make sure to follow Tamagui expo router starter and apply same changes!
as this tamagui starter is working fine, I'll close this issue as done

**Comment by Jankaz2** [comment]
Got the same problem when running the project on macos with version `1.90.14`. The same result after updating to the newest version `1.91.3`.  - Any workarounds?
On Windows it works with `1.90.14`.

**Comment by le-clucien** [comment]
Hello, also got the same problem using official tamagui `npm create tamagui@latest --template expo-router` command on macOS, I did it from fresh install in both macOS and an Alpine Linux distribution. 
I also tried to upgrade expo to the latest version but the issue persists.

Adding giornn0's workaround was the only way to make it running (on both iOS and Android), but as it was said, doing this might create other problems.

Maybe the problem isn't directly related to Tamagui starter but I don't think this issue should be closed as no viable solution emerged yet.

**Comment by le-clucien** [comment]
Hello, also got the same problem using official tamagui npm create tamagui@latest --template expo-router command on macOS, I did it from fresh install in both macOS and an Alpine Linux distribution.
I also tried to upgrade expo to the latest version but the issue persists.

Adding @giornn0's workaround was the only way to make it running (on both iOS and Android), but as it was said, doing this might create other problems.

Maybe the problem isn't directly related to Tamagui starter but I don't think this issue should be closed as no viable solution emerged yet.

**Comment by le-clucien** [comment]
Hello, also got the same problem using official tamagui npm create tamagui@latest --template expo-router command on macOS, I did it from fresh install in both macOS and an Alpine Linux distribution.
I also tried to upgrade expo to the latest version but the issue persists.

Adding @giornn0's workaround was the only way to make it running properly (on both iOS and Android), but as it was said, doing this might create other problems.

Maybe the problem isn't directly related to Tamagui starter but I don't think this issue should be closed as no viable solution emerged yet.

**Comment by ramraphael** [comment]
Bumping this as I'm encountering this issue when setting up Tamagui from scratch for a new React Native project at work. All configurations match what's on the Expo Starter.

**Comment by TheEhsanSarshar** [comment]
although I wasn't able to reproduce it with Tamagui official starter, yeah I'll check it again, maybe something is broken since then

**Comment by AdamDorwart** [comment]
Just confirming as a new user I just ran into this trying to add Tamagui to my existing Expo project. I followed the expo router getting started guide. I only see it on my native Android dev client (vs web).

For now, I will have to revert Tamagui from the project.

**Comment by ddikodroid** [comment]
I'm getting the error if I add `import '@tamagui/core/reset.css';` into `app/_layout.tsx`

If I delete that, the error is gone.

**Comment by kiprasmel** [comment]
@ddikodroid +1 on removing the css imports. 

this issue should be re-opened.

error i got before removing:

```console
iOS Bundling failed 9500ms (node_modules/expo-router/entry.js)
Unable to resolve "react-native-reanimated" from "node_modules/react-native-css-interop/dist/runtime/native/utils.js"
```

**Comment by saintpepsicola** [comment]
This is still an issue for me!

**Comment by TheEhsanSarshar** [comment]
There is a fix for windows users on master branch, and will be available with next release

**Comment by TheEhsanSarshar** [comment]
that fix is released now, you can upgrade your Tamagui version.
let me know if that fixed your issues

**Comment by herculesggimenes** [comment]
I'm still experiencing the same issue with the `import '../tamagui-web.css'` despite upgrading my Tamagui version

**Comment by FranciscoKloganB** [comment]
I know this is pretty frustrating for maintainers to ear, but, I am also getting this issue. Unfortunatly, I can't give you much context as I am new to React-Native/Expo/Tamagui ecosystem. I am doing a course and decide to use `tamagui` instead of `native-wind` because that is what I am looking forward to use in the future.

Like some users above, I only have this problem when I add `web/css` imports in `app/_layout.tsx`.

**Comment by Pamavoc** [comment]
Interested to know more about this solution too. I'm using expo@50.

**Comment by Andriiklymiuk** [comment]
same issue, 
```
"expo": "~51.0.17",
"tamagui": "^1.101.5"
```
does anyone know stable solution?

**Comment by likeSo** [comment]
Same issue here🙋
My workaround is comment this line in `app/_layout.tsx`:
```ts
// import '../tamagui-web.css';
```
I just upgraded to the latest `1.102.2` tamagui suite version, nothing changes.

expo sdk version is 51.0.21

**Comment by sandmule** [comment]
Adding  `jsxRuntime: 'automatic'` to the babel.config.js fixed it for me

```
    presets: [['babel-preset-expo', { jsxRuntime: 'automatic' }]],
```

**Comment by praneershrest** [comment]
Seeing the same issue when I try to import `import '../tamagui-web.css';`

Expo and tamagui versions:
```
"expo": "~51.0.26"
"tamagui": "^1.108.3"
```

Any updates regarding this issue?

**Comment by davidchicano** [comment]
**TL;DR**

I just solved the `ERROR There was a problem connecting to the react-native-css-interop Metro server` adding at the end of metro.config.js file:

````typescript
module.exports = config
````

**Explanation**

I had my project already created with default's Expo template, so I had to configure Tamagui from scratch. I followed the [official tamagui expo guide](https://tamagui.dev/docs/guides/expo). I did eeeverything exactly as says, but unfortunately I ran into that anoying metro error.

Reading your responses here, I saw some of you pointing out that the latest Tamagui expo template had no issues, so I just created one and compared with mine.

After adding [Tamagui Metro's config](https://tamagui.dev/docs/guides/metro#web-support) to my `metro.config.js`, it looked like this:

````typescript
// Learn more https://docs.expo.io/guides/customizing-metro
/**
 * @type {import('expo/metro-config').MetroConfig}
 */
const { getDefaultConfig } = require('expo/metro-config');

const config = getDefaultConfig(__dirname, {
  // [Web-only]: Enables CSS support in Metro.
  isCSSEnabled: true,
});

// Add Tamagui support with optimizing compiler + CSS extraction
const { withTamagui } = require('@tamagui/metro-plugin');
module.exports = withTamagui(config, {
  components: ['tamagui'],
  config: './tamagui.config.ts',
  outputCSS: './tamagui-web.css',
});
````
And the only difference with Tamagui's template were two lines:
````typescript
config.resolver.sourceExts.push('mjs')
module.exports = config
````
I don't know what they do nor where they come from, I just did some testing with one and another line, and I could prove that the solution was the `module.exports = config`. In case you wonder what I did with the other line `config.resolver.sourceExts.push('mjs')`, I am leaving it anyway just in case :)

Hope that helps you!

**Comment by rlee1990** [comment]
Doing what @davidchicano suggested worked for me

**Comment by Alaa-Ben** [comment]
FYI, a module can only have one export.

Adding `module.exports = config` just overrides `module.exports =  withTamagui(config, ...)`.
So it's not really a solution as it's basically just entirely removing Tamagi's metro config.

What you can try is only import `'../tamagui-web.css'; ` (app/_layout.tsx) on web (by using Platform.OS for example).

**Comment by rlee1990** [comment]
@Alaa-Ben your correct. My issue is when I add this I get an error 
```  An import declaration can only be used at the top level of a namespace or module.```
if I add the () after the word import I get another error.

**Comment by Alaa-Ben** [comment]
@rlee1990 That's because you're using `import`
Instead, try:
```
if (Platform.OS === 'web') {
  require('../tamagui-web.css')
}
```
